### PR TITLE
Update cartesian.js

### PR DESCRIPTION
**Pull request by chrisprice**

### CODE REVIEW

File: src/svg/cartesian.js [hunk 1]
```diff
@@ -1,206 +1 @@
-import { select, event } from 'd3-selection';
-import { scaleLinear } from 'd3-scale';
-import { seriesSvgLine } from 'd3fc-series';
-import { axisBottom, axisRight, axisLeft, axisTop } from 'd3fc-axis';
-import { dataJoin } from 'd3fc-data-join';
-import { rebindAll, exclude, prefix, includeMap } from 'd3fc-rebind';
 
-export default (xScale, yScale) => {
-    xScale = xScale || scaleLinear();
```
Comment (chrisprice) [inline src/svg/cartesian.js]: `scaleIdentity`?

File: src/svg/cartesian.js [hunk 2]
```diff
@@ -1,206 +1 @@
...
-    yScale = yScale || scaleLinear();
-
-    let yLabel = '';
-    let xLabel = '';
-    let yOrient = 'right';
```
Comment (chrisprice) [inline src/svg/cartesian.js]: My reading is that if these are changed after initial render, it'll not produce the right chart. If that's the case is it worth moving them to be factory arguments?

File: src/svg/cartesian.js [hunk 3]
```diff
@@ -1,206 +1 @@
...
-
-    const containerDataJoin = dataJoin('d3fc-group', 'cartesian-chart');
-
-    const cartesian = (selection) => {
-
-        selection.each((data, index, group) => {
-            const container = containerDataJoin(select(group[index]), [data]);
-
-            const yLabelElements = `<div class='y-axis-label' style='width: 1em; display: flex; align-items: center; justify-content: center'>
-                    <div class='label' style='transform: rotateZ(90deg)'></div>
```
Comment (chrisprice) [inline src/svg/cartesian.js]: `transform: rotate(90deg)`

File: src/svg/cartesian.js [hunk 4]
```diff
@@ -1,206 +1 @@
...
-                </div>`;
-            const yAxisElements = '<d3fc-svg class=\'y-axis\' style=\'width: 3em\'></d3fc-svg>';
```
Comment (chrisprice) [inline src/svg/cartesian.js]: Use backticks to avoid escaping?

File: src/svg/cartesian.js [hunk 5]
```diff
@@ -1,206 +1 @@
...
-
-            const xLabelElements = `<div class='x-axis-label' style='height: 1em; display: flex; align-items: center; justify-content: center'>
```
Comment (chrisprice) [inline src/svg/cartesian.js]: `text-align: center`?

File: src/svg/cartesian.js [hunk 6]
```diff
@@ -1,206 +1 @@
...
-                    <div class='label'></div>
-                </div>`;
-            const xAxisElements = '<d3fc-svg class=\'x-axis\' style=\'flex: 1\'></d3fc-svg>';
```
Comment (chrisprice) [inline src/svg/cartesian.js]: Backticks again

File: src/svg/cartesian.js [hunk 7]
```diff
@@ -1,206 +1 @@
...
-
-            const xAxisContainer = `<div style='height: 3em; display: flex; flex-direction: column; margin-${yOrient}: 4em'>
```
Comment (chrisprice) [inline src/svg/cartesian.js]: Could you use [`flex-direction: column-reverse`](https://developer.mozilla.org/en-US/docs/Web/CSS/flex-direction) to do the flip?

File: src/svg/cartesian.js [hunk 8]
```diff
@@ -1,206 +1 @@
...
-
-            container.enter()
-                .attr('style', 'display: flex; height: 100%; width: 100%; flex-direction: column')
-                .attr('auto-resize', '')
-                .html(`<div class='chart-label'
-                            style='height: ${chartLabel ? '2' : '0'}em; display: flex; align-items: center; justify-content: center'>
-                          <div class='label'></div>
-                      </div>
-                      ${xOrient === 'top' ? xAxisContainer : ''}
-                      <div style='flex: 1; display: flex; flex-direction: row'>
```
Comment (chrisprice) [inline src/svg/cartesian.js]: Same `flex-direction` trick?

File: src/svg/cartesian.js [hunk 9]
```diff
@@ -1,206 +1 @@
...
-            const xAxisContainer = `<div style='height: 3em; display: flex; flex-direction: column; margin-${yOrient}: 4em'>
-                    ${xOrient === 'bottom' ? xAxisElements : xLabelElements}
-                    ${xOrient === 'bottom' ? xLabelElements : xAxisElements}
-                </div>`;
-
-            container.enter()
-                .attr('style', 'display: flex; height: 100%; width: 100%; flex-direction: column')
-                .attr('auto-resize', '')
-                .html(`<div class='chart-label'
-                            style='height: ${chartLabel ? '2' : '0'}em; display: flex; align-items: center; justify-content: center'>
```
Comment (chrisprice) [inline src/svg/cartesian.js]: `text-align: center`?

File: src/svg/cartesian.js [hunk 10]
```diff
@@ -1,206 +1 @@
...
-                      <div style='flex: 1; display: flex; flex-direction: row'>
-                          ${yOrient === 'left' ? yLabelElements : ''}
-                          ${yOrient === 'left' ? yAxisElements : ''}
-                          <d3fc-svg class='plot-area' style='flex: 1; overflow: hidden'></d3fc-svg>
-                          ${yOrient === 'right' ? yAxisElements : ''}
-                          ${yOrient === 'right' ? yLabelElements : ''}
-                      </div>
-                      ${xOrient === 'bottom' ? xAxisContainer : ''}`);
-
-            container.select('.y-axis-label .label')
```
Comment (chrisprice) [inline src/svg/cartesian.js]: Any reason not to give the label a unique class name?

File: src/svg/cartesian.js [hunk 11]
```diff
@@ -1,206 +1 @@
...
-                .text(yLabel);
-
-            container.select('.x-axis-label .label')
-                .text(xLabel);
-
-            container.select('.chart-label .label')
-                .text(chartLabel);
-
-            select('.y-axis')
-                .on('resize', (d, i, nodes) => {
```
Comment (chrisprice) [inline src/svg/cartesian.js]: `measure`

File: src/svg/cartesian.js [hunk 12]
```diff
@@ -1,206 +1 @@
...
-                    if (yOrient === 'left') {
-                        const { width, height } = event.detail;
-                        select(nodes[i])
-                          .select('svg')
-                          .attr('viewBox', `${-width} 0 ${width} ${height}`);
```
Comment (chrisprice) [inline src/svg/cartesian.js]: You could just make the orient control the viewbox i.e. `${(yOrient === 'left') ? -width : 0} 0 ${width} ${height}`

Comment (chrisprice) [inline src/svg/cartesian.js]: I'm coming around to these changes allowing the orientations to be changed dynamically

File: src/svg/cartesian.js [hunk 13]
```diff
@@ -1,206 +1 @@
...
-                })
-                .on('draw', (d, i, nodes) => {
-                    select(nodes[i])
-                      .select('svg')
-                      .call(xAxis.scale(xScale));
-                });
-
-            container.select('.plot-area')
-                .on('resize', () => {
-                    const size = event.detail;
```
Comment (chrisprice) [inline src/svg/cartesian.js]: Consistency with above `const { width, height } = event.detail;`

File: src/svg/cartesian.js [hunk 14]
```diff
@@ -1,206 +1 @@
...
-        rebindAll(cartesian, xScale, includeMap({'ticks': 'xScaleTicks'}));
-    }
-    if (yScale.ticks) {
-        rebindAll(cartesian, yScale, includeMap({'ticks': 'yScaleTicks'}));
-    }
-
-    var axisExclusions = exclude(
-        'xScale', 'yScale'  // these are set by this components
-    );
-    rebindAll(cartesian, xAxis, axisExclusions, prefix('x'));
```
Comment (chrisprice) [inline src/svg/cartesian.js]: The axis is rebound here but changes below...

File: src/svg/cartesian.js (hunk 7 above)
Comment (ColinEberhardt) [inline src/svg/cartesian.js]: Nice - I had not idea flexbox has reverse orientations!

**Comment by ColinEberhardt** [comment]
Thanks for the review - I've made a bunch of changes this branch:

https://github.com/ColinEberhardt/d3fc-chart/tree/updates

The use of reverse flexbox has made the code much simpler. I've kept the axis / scale rebind logic - I couldn't find a suitable subset of properties to expose, without making some quite arbitrary choices.

**Pull request closed without merging**